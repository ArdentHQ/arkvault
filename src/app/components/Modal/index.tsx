export * from "./hooks";
export * from "./Modal";
export * from "./Modal.blocks";
