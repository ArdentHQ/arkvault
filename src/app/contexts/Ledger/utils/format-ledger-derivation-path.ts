import { LedgerDerivationScheme } from "@/app/contexts/Ledger/Ledger.contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
