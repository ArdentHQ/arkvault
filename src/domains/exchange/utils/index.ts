export * from "./assertions";
export * from "./errors";
