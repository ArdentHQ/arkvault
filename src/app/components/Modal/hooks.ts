import { useCallback, useEffect, useMemo } from "react";
import { browser } from "@/utils/platform";

export const useModal = ({ isOpen, onClose }: { isOpen: boolean; onClose?: any }) => {
	const defaultOverflow = useMemo(() => (browser.supportsOverflowOverlay() ? "overlay" : "scroll"), []);

	useEffect(() => {
		document.body.style.overflowY = defaultOverflow;

		if (isOpen) {
			document.body.style.overflowY = "hidden";
		}

		return () => {
			document.body.style.overflowY = defaultOverflow;
		};
	}, [defaultOverflow, isOpen]);

	const onEscKey = useCallback(
		(event: KeyboardEvent) => {
			event.preventDefault();
			event.stopPropagation();
			if (event.key === "Escape") {
				onClose();
			}
		},
		[onClose],
	);

	useEffect(() => {
		if (isOpen) {
			document.addEventListener("keyup", onEscKey, false);
		} else {
			document.removeEventListener("keyup", onEscKey);
		}

		return () => {
			document.removeEventListener("keyup", onEscKey);
		};
	}, [isOpen, onEscKey]);
};
