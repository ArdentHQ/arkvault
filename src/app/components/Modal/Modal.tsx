import React, { useEffect, useRef } from "react";

import { useModal } from "./hooks";
import { ModalContainer, ModalContent } from "./Modal.blocks";
import { Size } from "@/types";
import { useNavigationContext } from "@/app/contexts";
import { DefaultTFuncReturn } from "i18next";

interface ModalProperties extends JSX.IntrinsicAttributes {
	children: React.ReactNode;
	title: string | React.ReactNode;
	titleClass?: string;
	description?: string | JSX.Element | DefaultTFuncReturn;
	banner?: React.ReactNode;
	image?: React.ReactNode;
	noButtons?: boolean;
	hideCloseButton?: boolean;
	size?: Size;
	isOpen: boolean;
	onClose?: any;
	onClick?: any;
}

const Modal = ({
	isOpen,
	description,
	title,
	titleClass,
	banner,
	image,
	noButtons,
	hideCloseButton,
	size,
	children,
	onClose,
	...attributes
}: ModalProperties) => {
	const referenceShouldClose = useRef<boolean>();
	const { setShowMobileNavigation } = useNavigationContext();

	useModal({ isOpen, onClose });

	useEffect(() => {
		if (isOpen) {
			setShowMobileNavigation(false);
		} else {
			setShowMobileNavigation(true);
		}

		return () => setShowMobileNavigation(true);
	}, [isOpen]);

	if (!isOpen) {
		return <></>;
	}

	const handleClickOverlay = (event: React.MouseEvent<HTMLElement>) => {
		if (referenceShouldClose.current === undefined) {
			referenceShouldClose.current = true;
		}

		if (!referenceShouldClose.current) {
			referenceShouldClose.current = undefined;
			return;
		}

		event.preventDefault();
		event.stopPropagation();
		onClose?.();
	};

	const handleClickContent = () => {
		referenceShouldClose.current = false;
	};

	return (
		<div
			className="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
			onClick={handleClickOverlay}
			data-testid="Modal__overlay"
			{...attributes}
		>
			<ModalContainer
				size={size}
				onMouseDown={handleClickContent}
				onMouseUp={handleClickContent}
				onClick={handleClickContent}
				tabIndex={-1}
			>
				<ModalContent
					aria-selected={isOpen}
					title={title}
					titleClass={titleClass}
					description={description}
					banner={banner}
					image={image}
					onClose={onClose}
					hideCloseButton={hideCloseButton}
					noButtons={noButtons}
				>
					{children}
				</ModalContent>
			</ModalContainer>
		</div>
	);
};

Modal.displayName = "Modal";

export { Modal };
