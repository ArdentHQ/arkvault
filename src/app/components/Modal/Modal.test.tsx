/* eslint-disable @typescript-eslint/require-await */
import userEvent from "@testing-library/user-event";
import React from "react";

import { Modal } from "./Modal";
import { render, screen, renderResponsive, waitFor } from "@/utils/testing-library";
import { browser } from "@/utils/platform";

describe("Modal", () => {
	it("should not render if not open", () => {
		const { asFragment } = render(<Modal title="ark" isOpen={false} />);

		expect(screen.queryByTestId("Modal__inner")).not.toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it.each([true, false])("should render a modal with overlay", (supportsOverflowOverlay) => {
		const overflowOverlayMock = vi
			.spyOn(browser, "supportsOverflowOverlay")
			.mockReturnValue(supportsOverflowOverlay);

		const { asFragment } = render(
			<Modal title="ark" isOpen={true}>
				This is the Modal content
			</Modal>,
		);

		expect(screen.getByTestId("Modal__overlay")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();

		overflowOverlayMock.mockRestore();
	});

	it("should render with scrollbar if it exceeds window height", () => {
		const windowHeight = window.innerHeight;
		Object.defineProperty(window, "innerHeight", { value: 40 });

		const { asFragment } = render(
			<Modal title="ark" isOpen={true}>
				This is the Modal content
			</Modal>,
		);

		expect(asFragment()).toMatchSnapshot();
		Object.defineProperty(window, "innerHeight", { value: windowHeight });
	});

	it("should render without offset for buttons", () => {
		const { asFragment } = render(
			<Modal title="ark" isOpen={true} noButtons>
				This is the Modal content
			</Modal>,
		);

		expect(screen.getByTestId("Modal__inner")).toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render without close button", () => {
		const { asFragment } = render(
			<Modal title="ark" isOpen={true} hideCloseButton>
				This is the Modal content
			</Modal>,
		);

		expect(screen.queryByTestId("Modal__close-button")).not.toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();
	});

	it("should closed by click on overlay", () => {
		const onClose = vi.fn();
		render(
			<Modal title="ark" isOpen={true} onClose={onClose}>
				This is the Modal content
			</Modal>,
		);

		expect(screen.getByTestId("Modal__overlay")).toBeInTheDocument();
		expect(screen.getByTestId("Modal__inner")).toBeInTheDocument();

		userEvent.click(screen.getByTestId("Modal__overlay"));

		expect(onClose).toHaveBeenCalledWith();
	});

	it("should no close by click on modal content", () => {
		const onClose = vi.fn();
		render(
			<Modal title="ark" isOpen={true} onClose={onClose}>
				This is the Modal content
			</Modal>,
		);

		expect(screen.getByTestId("Modal__overlay")).toBeInTheDocument();
		expect(screen.getByTestId("Modal__inner")).toBeInTheDocument();

		userEvent.click(screen.getByTestId("Modal__inner"));

		expect(onClose).not.toHaveBeenCalled();
	});

	it("should closed by the Esc key", async () => {
		const onClose = vi.fn();
		const { asFragment } = render(<Modal title="ark" isOpen={true} onClose={onClose} />);

		expect(screen.getByTestId("Modal__overlay")).toBeInTheDocument();

		userEvent.keyboard("{enter}");

		expect(onClose).not.toHaveBeenCalled();

		userEvent.keyboard("{esc}");

		expect(onClose).toHaveBeenCalledTimes(1);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render a modal with description", () => {
		const { asFragment } = render(<Modal title="ark" description="This is the Modal description" isOpen={true} />);

		expect(asFragment()).toMatchSnapshot();
		expect(screen.getByTestId("Modal__inner")).toHaveTextContent("This is the Modal description");
	});

	it("should render a modal with content", () => {
		const { asFragment } = render(
			<Modal title="ark" isOpen={true}>
				This is the Modal content
			</Modal>,
		);

		expect(asFragment()).toMatchSnapshot();
		expect(screen.getByTestId("Modal__inner")).toHaveTextContent("This is the Modal content");
	});

	it("should render a small one", () => {
		const { container } = render(<Modal title="ark" size="sm" isOpen={true} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a medium one", () => {
		const { container } = render(<Modal title="ark" size="md" isOpen={true} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a large one", () => {
		const { container } = render(<Modal title="ark" size="lg" isOpen={true} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a xlarge one", () => {
		const { container } = render(<Modal title="ark" size="xl" isOpen={true} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a 3x large one", () => {
		const { container } = render(<Modal title="ark" size="3xl" isOpen={true} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a 4x large one", () => {
		const { container } = render(<Modal title="ark" size="4xl" isOpen={true} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a 5x large one", () => {
		const { container } = render(<Modal title="ark" size="5xl" isOpen={true} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a modal with banner", () => {
		const { asFragment } = render(
			<Modal title="ark" isOpen={true} banner={true}>
				This is the Modal content
			</Modal>,
		);

		expect(screen.getByTestId("Modal__overlay")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should hide the mobile menu when the modal is opened", async () => {
		renderResponsive(
			<Modal title="ark" isOpen={true}>
				This is the Modal content
			</Modal>,
			"xs",
		);

		await waitFor(() => expect(screen.queryByTestId("NavigationBarMobile")).not.toBeInTheDocument());
	});
});
