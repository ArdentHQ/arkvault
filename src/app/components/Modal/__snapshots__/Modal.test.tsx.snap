// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should closed by the Esc key 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1nvh1sc"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal should not render if not open 1`] = `<DocumentFragment />`;

exports[`Modal should render a 3x large one 1`] = `
<div>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1bkh76g"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should render a 4x large one 1`] = `
<div>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-pzbet3"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should render a 5x large one 1`] = `
<div>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-11haine"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should render a large one 1`] = `
<div>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1vhnc7q"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should render a medium one 1`] = `
<div>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-4e0zgx"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should render a modal with banner 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1nvh1sc"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <div
            class="relative -mx-10 mb-10 -mt-10 h-56"
          >
            <div
              class="absolute bottom-0 left-0 mb-10 ml-10"
            >
              <h2
                class="m-0 text-4xl font-extrabold leading-tight text-theme-text"
              >
                ark
              </h2>
            </div>
          </div>
          <div
            class="flex-1"
          >
            This is the Modal content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal should render a modal with content 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1nvh1sc"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          >
            This is the Modal content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal should render a modal with description 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1nvh1sc"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          >
            <div
              class="whitespace-pre-line text-theme-secondary-text"
            >
              This is the Modal description
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal should render a modal with overlay 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1nvh1sc"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          >
            This is the Modal content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal should render a modal with overlay 2`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1nvh1sc"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          >
            This is the Modal content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal should render a small one 1`] = `
<div>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-xrj5ng"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should render a xlarge one 1`] = `
<div>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1icbhox"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should render without close button 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1nvh1sc"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          >
            This is the Modal content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal should render without offset for buttons 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1nvh1sc"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl p-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            ark
          </h2>
          <div
            class="flex-1"
          >
            This is the Modal content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
